oauth2-server/src/main/java/com/clouway/oauth2/CodeExchangeVerificationFlow.java

```diff
@@ -0,0 +1,30 @@
+package com.clouway.oauth2;
+
+import com.clouway.friendlyserve.Request;
+import com.clouway.friendlyserve.Response;
+import com.clouway.oauth2.authorization.Authorization;
+import com.clouway.oauth2.client.Client;
+import com.clouway.oauth2.codechallenge.CodeVerifier;
+
+/**
+ * @author Miroslav Genov (miroslav.genov@clouway.com)
+ */
+class CodeExchangeVerificationFlow implements AuthorizedClientActivity {
+  private final CodeVerifier codeVerifier;
+  private final AuthorizedClientActivity authorizedClientActivity;
+
+  CodeExchangeVerificationFlow(CodeVerifier codeVerifier, AuthorizedClientActivity authorizedClientActivity) {
+    this.codeVerifier = codeVerifier;
+    this.authorizedClientActivity = authorizedClientActivity;
+  }
+
+  @Override
+  public Response execute(Authorization authorization, Client client, Request request, DateTime instant) {
+    String codeVerifierValue = request.param("code_verifier") == null ? "" : request.param("code_verifier");
```

Comment:
Here you ensure that code_verifier is empty string or some value but in the verifier you are checking for null values which smells.

Reply:
I agree that I've over-checked for null. Removed the check.

Reply:
Also added a test that shows that if it's not provided in the request an empty value will be provided.